rust/garbage_collector/src/operators/delete_unused_files.rs

```diff
@@ -258,21 +252,13 @@ impl Operator<DeleteUnusedFilesInput, DeleteUnusedFilesOutput> for DeleteUnusedF
             }
             CleanupMode::Delete => {
                 // Hard delete - remove the file
-                let mut futures = Vec::new();
-                for file_path in &all_files {
-                    futures.push(self.delete_with_path(file_path.clone()));
-                }
-
-                let num_futures = futures.len();
-                if num_futures > 0 {
-                    let results: Vec<Result<(), FileOperationError>> =
-                        futures::stream::iter(futures)
-                            .buffer_unordered(num_futures)
-                            .collect()
-                            .await;
+                if !all_files.is_empty() {
+                    let mut delete_stream = futures::stream::iter(all_files.clone())
+                        .map(move |file_path| self.delete_with_path(file_path))
+                        .buffer_unordered(100);
```

Comment:
greatly reduces peak memory usage when GC'ing a large set of files--went from a peak of > 512MB (was OOMing) to a peak < 75MB

I don't have a principled reason for setting this to 100. Seems to work well for now though. I would be open to changing this to use the admission controlled S3 impl, just unsure if it makes sense to use the same admission controller for uploads/downloads and deletes.